lib/protobuf.php

```diff
@@ -354,11 +354,19 @@ public function writeBool(bool $b): void {
     }
 
     public function writeFloat(float $f): void {
-      $this->buf .= \pack('f', $f);
+      // TODO: Compute the position before growing the string.
+      // Unfortunately this crashes on HHVM 4.153.1 (but not on HHVM 4.164.0).
+      $this->buf .= "\x0\x0\x0\x0";
+      $pos = \strlen($this->buf) - 4;
+      /*HH_FIXME[4107]*//*HH_FIXME[2049]*/\HH\set_bytes_float32($this->buf, $pos, $f);
```

Comment:
Unfortunately the consensus is that we can't do better than those 2 `HH_FIXME`s.